www/src/components/sidebar/item.js

```diff
@@ -13,61 +13,73 @@ const isItemActive = (activeItemParents, item) => {
   return false
 }
 
-class Item extends React.PureComponent {
-  render() {
-    const {
-      activeItemLink,
-      activeItemParents,
-      isActive,
-      openSectionHash,
-      item,
-      location,
-      onLinkClick,
-      onSectionTitleClick,
-      ui,
-      isSingle,
-      disableAccordions,
-    } = this.props
+const Item = ({
+  activeItemLink,
+  activeItemParents,
+  isActive,
+  openSectionHash,
+  item,
+  location,
+  onLinkClick,
+  onSectionTitleClick,
+  ui,
+  isSingle,
+  disableAccordions,
+}) => {
+  const itemRef = useCallback(
+    async node => {
+      if (item.link === activeItemLink.link && node !== null) {
+        // this noop for whatever reason gives time for React to know what
+        // ref is attached to the node to scroll to it, removing this line
+        // will only scroll to the correct location on a full page refresh,
+        // instead of navigating between pages with the prev/next buttons
+        // or clicking on linking guides or urls
+        await function() {}
```

Comment:
I left this comment to try and explain this but I'm honestly not super sure why this doesn't work without this. I solicited help on this but never heard from anyone else about it so just went with the best solution I could come up with 🙂